classes/publication/DAO.php

```diff
@@ -56,4 +65,77 @@ public function fromRow(object $primaryRow): Publication
+            )
+            ->when($title != null, fn (Builder $q) => $q->leftJoin('publication_settings AS pst', 'p.publication_id', '=', 'pst.publication_id'))
+            ->when(
+                $author != null,
+                fn (Builder $q) => $q->leftJoin('authors AS au', 'p.publication_id', '=', 'au.publication_id')
+                    ->leftJoin(
+                        'author_settings AS asgs',
+                        fn (JoinClause $j) => $j->on('asgs.author_id', '=', 'au.author_id')
+                            ->where('asgs.setting_name', '=', Identity::IDENTITY_SETTING_GIVENNAME)
+                    )
+                    ->leftJoin(
+                        'author_settings AS asfs',
+                        fn (JoinClause $j) => $j->on('asfs.author_id', '=', 'au.author_id')
+                            ->where('asfs.setting_name', '=', Identity::IDENTITY_SETTING_FAMILYNAME)
+                    )
+            )
+            ->when(
+                $pubIdSettingName,
+                fn (Builder $q) => $q->leftJoin(
+                    'submission_settings AS pss',
+                    fn (JoinClause $j) => $j->on('s.submission_id', '=', 'pss.submission_id')
+                        ->where('pss.setting_name', '=', $pubIdSettingName)
+                )
+            )
+            ->where('s.status', '=', Submission::STATUS_PUBLISHED)
+            ->where('s.context_id', '=', $contextId)
+            ->where('p.version_stage', '=', VersionStage::VERSION_OF_RECORD)
+            ->where('p.status', '=', Submission::STATUS_PUBLISHED) // TO-DO: see if this is enough, or if we should consider published or date_published column ?
+            ->whereNull('p2.publication_id')
+            ->when($pubIdType != null, fn (Builder $q) => $q->where('pspidt.setting_name', '=', "pub-id::{$pubIdType}")->whereNotNull('pspidt.setting_value'))
+            ->when($title != null, fn (Builder $q) => $q->where('pst.setting_name', '=', 'title')->where('pst.setting_value', 'LIKE', "%{$title}%"))
+            ->when($author != null, fn (Builder $q) => $q->whereRaw("CONCAT(COALESCE(asgs.setting_value, ''), ' ', COALESCE(asfs.setting_value, '')) LIKE ?", [$author]))
+            ->when($issueId != null, fn (Builder $q) => $q->where('p.issue_id', '=', $issueId))
+            ->when(
+                $pubIdSettingName,
+                fn (Builder $q) => $q->when(
+                    $pubIdSettingValue === null,
+                    fn (Builder $q) => $q->whereRaw("COALESCE(pss.setting_value, '') = ''"),
+                    fn (Builder $q) => $q->when(
+                        $pubIdSettingValue != PubObjectsExportPlugin::EXPORT_STATUS_NOT_DEPOSITED,
+                        fn (Builder $q) => $q->where('pss.setting_value', '=', $pubIdSettingValue),
+                        fn (Builder $q) => $q->whereNull('pss.setting_value')
+                    )
+                )
+            )
+            ->orderByDesc('s.submission_id')
```

Comment:
Could this be missing a groupBy, e.g. `->groupBy('p.publication_id')`? It seems to fix the count and display issue on the DOAJ plugin as well.